test/source/mock/attester/attester-endpoints.ts

```diff
@@ -1,437 +1,87 @@
-    if (isGet(req)) {
-      if (emailOrLongid === 'test.ldap.priority@gmail.com') {
-        return protonMailCompatKey;
-      }
-      if (emailOrLongid === 'test.flowcrypt.pubkeyserver.priority@gmail.com') {
-        return protonMailCompatKey;
-      }
-      if (emailOrLongid === 'has.pub@client-configuration-test.flowcrypt.test') {
-        return hasPubKey;
-      }
-      if (emailOrLongid === 'invalid.pub@client-configuration-test.flowcrypt.test') {
-        return protonMailCompatKey;
-      }
-      if (emailOrLongid === 'test.ldap.keyserver.pgp@gmail.com' && server === 'keyserver.pgp.com') {
-        return [protonMailCompatKey, testMatchPubKey].join('\n');
-      }
-      if (emailOrLongid === 'test.ldap.timeout@gmail.com') {
-        throw new HttpClientErr('RequestTimeout', Status.BAD_REQUEST);
-      }
-      if (emailOrLongid === 'test.flowcrypt.pubkey.timeout@gmail.com') {
-        return somePubkey;
-      }
-      throw new HttpClientErr('No OpenPGP LDAP server on this address.', Status.NOT_FOUND);
-    } else {
-      throw new HttpClientErr(`Not implemented: ${req.method}`);
-    }
-  },
-  '/attester/welcome-message': async ({ body }, req) => {
-    if (!isPost(req)) {
-      throw new HttpClientErr(`Wrong method: ${req.method}`);
-    }
-    const { email, pubkey } = body as Dict<string>;
-    expect(email).to.contain('@');
-    expect(pubkey).to.contain('-----BEGIN PGP PUBLIC KEY BLOCK-----');
-    return { sent: true };
-  },
+interface PubKeyLookUpResult {
+  pubkey?: string;
+  delayInSeconds?: number;
+  domainToCheck?: string;
+  returnError?: { code: number; message: string };
+}
+
+export interface AttesterConfig {
+  pubkeyLookup?: Record<string, PubKeyLookUpResult>;
+  ldapRelay?: Record<string, PubKeyLookUpResult>;
+  welcomeMessageEnabled?: boolean;
+}
+
+export const getMockAttesterEndpoints = (attesterConfig: AttesterConfig): HandlersDefinition => {
+  return {
+    '/attester/pub/?': async ({ body }, req) => {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const emailOrLongid = req.url!.split('/').pop()!.toLowerCase().trim();
+      if (isGet(req)) {
+        if (!attesterConfig?.pubkeyLookup) {
+          throw new HttpClientErr('Method not allowed', 405);
+        }
+        const pubRes = attesterConfig.pubkeyLookup[emailOrLongid];
+        if (pubRes) {
+          if (pubRes.returnError) {
+            throw new HttpClientErr(pubRes.returnError.message, pubRes.returnError.code);
+          }
+          if (pubRes.delayInSeconds) {
+            await Util.sleep(pubRes.delayInSeconds);
+          }
+          return pubRes.pubkey;
+        }
+        throw new HttpClientErr('Pubkey not found', 404);
+      } else if (isPost(req)) {
+        oauth.checkAuthorizationForEmail(req.headers.authorization, emailOrLongid);
+        expect(body).to.contain('-----BEGIN PGP PUBLIC KEY BLOCK-----');
+        MOCK_ATTESTER_LAST_INSERTED_PUB[emailOrLongid] = body as string;
```

Comment:
The tests will continue to interfere with each other via this global output constant.
Should the reference to the output be included in the config?

Reply:
Yes, I have also considered this issue but haven't found a suitable solution yet. At the moment, this POST method is only used in two places, and they are attempting with different email addresses, so it's not a problem for now. However, it could potentially become a problem in the future.

Could you please clarify what you mean by "reference to the output being included in the config"?

Reply:
What about not using separate `MOCK_ATTESTER_LAST_INSERTED_PUB` object, but add new keys directly to `attesterConfig.pubkeyLookup`, like this `attesterConfig.pubkeyLookup[emailOrLongid] = { pubkey: body as string };`?

Then in tests we can check value of `t.mockApi.attesterConfig.pubkeyLookup![recipient]`

Reply:
Pushed update. Please check